internal/credential/attributes.go

```diff
@@ -122,8 +122,6 @@ func GetCredentialsConfig(secrets *structpb.Struct, attrs *CredentialAttributes)
 	// static credentials requires the private key and client email
 	case privateKey != "" && attrs.ClientEmail == "":
 		badFields[fmt.Sprintf("attributes.%s", ConstClientEmail)] = "must not be empty when private key is set"
-	case attrs.ClientEmail != "" && privateKey == "":
```

Comment:
So I run into an issue where UpdateHostCatalog can be called without passing in new secrets. In this case, the attribute can have `client_email` set but no `private_key` is passed. When we call it here: https://github.com/hashicorp/boundary-plugin-gcp/blob/cf0a591dcbe24fcc2c000e77200acf2f261dfea9/plugin/service/host/plugin.go#L155. 

It fails because it expected a private key